specification/machinelearningservices/data-plane/Microsoft.MachineLearningServices/preview/2024-05-01-preview/azure-ai-assets.json

```diff
@@ -0,0 +1,1027 @@
+        "value"
+      ]
+    },
+    "Prompt": {
+      "type": "object",
+      "description": "Prompt resource definition",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Fully qualified resource Id: azureml://workspace/{workspaceName}/indexes/{name}/versions/{version} of the index.",
+          "readOnly": true
+        },
+        "stage": {
+          "type": "string",
+          "description": "Update stage to 'Archive' for soft delete. Default is Development, which means the asset is under development.",
+          "default": "Development"
+        },
+        "description": {
+          "type": "string",
+          "description": "Description information of the asset."
+        },
+        "systemData": {
+          "$ref": "#/definitions/SystemData",
+          "description": "Metadata containing createdBy and modifiedBy information.",
+          "readOnly": true
+        },
+        "tags": {
+          "type": "object",
+          "description": "Asset's tags.",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "properties": {
+          "type": "object",
+          "description": "Asset's properties.",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "dataUri": {
+          "type": "string",
+          "description": "Default workspace blob storage Ui. Should work across storage types and auth scenarios."
+        },
+        "templatePath": {
+          "type": "string",
+          "description": "Relative path of the prompt data file at the dataUri location"
+        }
+      },
+      "required": [
+        "id",
+        "stage",
+        "dataUri",
+        "templatePath"
+      ]
+    },
+    "SystemData": {
+      "type": "object",
+      "description": "Metadata pertaining to creation and last modification of the resource.",
+      "properties": {
+        "createdAt": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The timestamp the resource was created at.",
+          "readOnly": true
+        },
+        "createdBy": {
+          "type": "string",
+          "description": "The identity that created the resource.",
+          "readOnly": true
+        },
+        "createdByType": {
+          "type": "string",
+          "description": "The identity type that created the resource.",
+          "readOnly": true
+        },
+        "lastModifiedAt": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The timestamp of resource last modification (UTC)",
+          "readOnly": true
+        }
+      }
+    },
+    "VersionInfo": {
+      "type": "object",
+      "description": "Next version definition.",
+      "properties": {
+        "nextVersion": {
+          "type": "integer",
+          "format": "int64",
+          "description": "Next version as defined by the server."
+        },
```

Comment:
I recommended above that you simply delete this property, and I think that's the best course of action. But if you do keep it, I think you need to make it a string. It's just weird that version is a string everywhere else in the API except here.

Reply:
For this case in particular, when we increment the next version we do a +1 on the previous version so for this case we assume version must be an integer vs string

Reply:
But all the previous versions are strings. So what if the version is "one" or "my-favorite-version"? This is extremely ill-defined and will only create problems down the road. Please remove it.